commands/secret_test.go

```diff
@@ -285,6 +285,66 @@ func TestHandleSecretSearchCmd(t *testing.T) {
 	}
 }
 
+func TestHandleSecretServiceprincipalSearchCmd(t *testing.T) {
+	testCase := []struct {
+		name        string
+		args        string
+		apiResponse []byte
+		out         []byte
+		expectedErr *errors.ApiError
+	}{
+		{
+			name:        "Happy path",
+			args:        "sp",
+			apiResponse: []byte(`test`),
+			out:         []byte(`test`),
+			expectedErr: nil,
+		},
+		{
+			name:        "api Error",
+			args:        "sp",
+			apiResponse: []byte(`test`),
+			out:         []byte(`test`),
+			expectedErr: errors.NewS("error"),
+		},
+	}
+
```

Comment:
Missing t.Parallel() in both outer and child scope. Can you add that please unless specific reason it can't run that way?

Also in the other for each test case and calling scope.